### ISSUE TITLE
Implement human annotation/labeling system for evaluation workflows

### ISSUE THREAD

**Issue by snopoke** [dimagi/open-chat-studio#2682]
## Problem

We have automated evaluations (`apps/evaluations`) but no way to collect structured human feedback on bot responses. Current session/message tagging is insufficient for:
- Systematic quality audits requiring multiple reviewers
- Creating gold-standard datasets for evaluator training
- Measuring inter-annotator agreement
- Queuing and tracking annotation work

### Core Features

1. **Annotation Queues**
   - Queues are assignable to specific users
   - Each queue has a customizable annotation schema (what data to collect)
   - Configurable N-reviews-per-item requirement (1-10 reviews before item is complete)
   - Track queue progress and completion status

2. **Customizable Annotation Schemas**
   - Reuse existing `FieldDefinition` pattern from evaluators (`apps/evaluations/field_definitions.py`)
   - Support field types: choice, int, float, string
   - Validation constraints (min/max, required fields, etc.)
   - Schema templates for common use cases (quality review, safety review, intent classification)

3. **Annotation Items**
   - Items can be: chat messages, full sessions, or external data (CSV imports)
   - Display relevant context to annotators (conversation history, metadata, participant info)
   - Priority ordering (high-priority items shown first)
   - Status tracking: pending → in progress → completed → flagged

4. **Multi-Reviewer Workflow**
   - Admin sets required number of reviews per item (e.g., "needs 3 reviews")
   - System prevents duplicate annotation (same user can't review same item twice)
   - Item marked complete when required review count reached
   - Queue completion triggers when all items complete

5. **Annotator Experience**
   - Simple, focused UI: one item at a time
   - Form auto-generated from schema (similar to how evaluator forms work)
   - Actions: Save draft, Submit, Skip/Flag for review
   - "Next item" logic: get oldest unreviewed item assigned to this user
   - Progress indicator: "Item 23 of 100"
   - Keyboard shortcuts for fast annotation

6. **Admin Dashboard**
   - Create queues: select schema, assign users, set review requirements
   - Bulk add items:
     - From chat sessions (filter by tags, date range, experiment)
     - From evaluation results (e.g., "queue all results where confidence < 0.5")
     - From CSV upload
   - View queue progress and annotator stats
   - Export annotations (CSV/JSONL)

### Advanced Features (Phase 2+)

7. **Quality Control**
   - Gold standard items (known correct answers) for quality checks
   - Track annotator accuracy on gold standards
   - Inter-annotator agreement metrics (Cohen's kappa, percent agreement)
   - Flag low-agreement items for expert review

8. **Integration with Evaluations**
   - Create annotation queues directly from evaluation results
   - Convert completed annotations into `EvaluationDataset` for training evaluators
   - Use human labels as ground truth for evaluator validation

### Data Model

```
AnnotationSchema
  - name, description
  - fields: Dict[str, FieldDefinition]  # Reuse existing pattern
  
AnnotationQueue
  - name
  - schema (FK)
  - assigned_users (M2M)
  - required_reviews_per_item (int, 1-10)
  - status (active/paused/completed)
  - enable_agreement_checks (bool)
  
AnnotationItem
  - queue (FK)
  - chat_message (FK, nullable)  # Link to source data
  - experiment_session (FK, nullable)
  - data (JSON)  # For external/synthetic data
  - display_text (text)
  - context (JSON)  # History, metadata for display
  - status (pending/in_progress/completed/flagged)
  - priority (int)
  - is_gold_standard (bool)
  - gold_standard_annotation (JSON)
  
Annotation
  - item (FK)
  - annotator (FK to User)
  - data (JSON)  # The actual annotation, validated against schema
  - status (draft/submitted/rejected)
  - time_spent_seconds (int)
  - matches_gold_standard (bool, nullable)
  
AnnotatorStats (optional, Phase 2)
  - user, queue
  - total_annotations, avg_time
  - gold_standard_accuracy, avg_agreement_score
```

### UI Flow

**Annotator:**
1. `/annotations/queues/` - List assigned queues with progress
2. `/annotations/queue/<id>/annotate/` - Annotation interface
   - Left: Item display (message + context)
   - Right: Form (auto-generated from schema)
   - Actions: Save draft / Submit / Skip
3. `/annotations/my-history/` - View completed annotations

**Admin:**
1. `/annotations/admin/` - Queue management (CRUD)
2. `/annotations/queue/<id>/dashboard/` - Progress, stats, export
3. `/annotations/queue/<id>/items/` - Item management
4. `/annotations/item/<id>/` - Compare multiple annotations, resolve conflicts

### Integration Points

- **Evaluation results → Annotation queue**: Button to "Send low-confidence results to human review"
- **Annotations → Evaluation dataset**: Convert completed annotations to `EvaluationDataset` for evaluator training
- **Session tagging**: Bulk-add sessions with specific tags to annotation queues
- **Export formats**: CSV for analysis, JSONL for fine-tuning datasets

### Phase 1: Core Annotation (MVP)
- [ ] `AnnotationSchema`, `AnnotationQueue`, `AnnotationItem`, `Annotation` models
- [ ] Admin UI: Create queue, assign users, bulk-add items from sessions
- [ ] Annotator UI: Simple form view with next/submit/skip
- [ ] Basic export (CSV)

### Phase 2: Multi-Reviewer & Quality
- [ ] N-reviewer completion logic
- [ ] Gold standard items and accuracy tracking
- [ ] Annotator stats dashboard
- [ ] Inter-annotator agreement metrics

### Phase 3: Advanced Integration
- [ ] Create annotation queues from evaluation results
- [ ] Convert annotations to evaluation datasets
- [ ] Comparison/conflict resolution UI
- [ ] API endpoints for external tools

## Acceptance Criteria

- Admin can create an annotation queue with custom schema, assign 3 users, require 2 reviews per item
- Annotator sees only items they haven't reviewed yet
- Item marked complete after N reviews submitted
- Queue marked complete when all items complete
- Annotations export to CSV with all schema fields as columns
- System prevents same user from reviewing same item twice

## References

- Existing pattern: `apps/evaluations/` (especially `field_definitions.py`, `models.py`)
- Similar tools: Label Studio, Argilla, LangSmith annotations
